### PR TITLE
mouseless: update desc and depends_on macos

### PR DESCRIPTION
**Pull request by cymian**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---

Multiple users have reported the app not working on macOS 13 and below.  So I've updated the required macos version to `sonoma`.  Also tweaked the description a bit.

### CODE REVIEW

**Comment by croian** [comment]
note, I'm the developer of the app (and maintainer of https://github.com/croian/mouseless-issues)

File: Casks/m/mouseless.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
   url "https://mouseless.click/mouseless-installer_v#{version}.dmg"
   name "mouseless"
-  desc "Mouse control with the keyboard"
+  desc "Full, fast mouse control with the keyboard"
```
Comment (bevanjkay) [inline Casks/m/mouseless.rb:7]:
```suggestion
  desc "Mouse control with the keyboard"
```
Per our `desc` stanza guidelines, we trim the marketing fluff out of descriptions in `homebrew-cask`.
https://docs.brew.sh/Cask-Cookbook#stanza-desc

**Review by bevanjkay — COMMENTED** [review]
Thanks @croian - just a note below about reverting the `desc` changes.

If users are having issues with specific system versions, I would recommend including that in your app's `info.plist` file too.

```
<key>LSMinimumSystemVersion</key>
<string>14.0</string>
```

This would prevent MacOS from opening that app on older versions.

File: Casks/m/mouseless.rb (hunk 1 above)
Comment (croian) [reply Casks/m/mouseless.rb]: understandable, thank you for the link, will follow those from now on.  I asked claude if it was appropriate, but apparently he's not up to date on those guidelines.

**Review by bevanjkay — APPROVED** [review]

**Comment by croian** [comment]
thank you for the tip about the plist @bevanjkay, that is super useful to know

**Pull request merged**